src/Chainweb/Utils.hs

```diff
@@ -877,16 +878,19 @@ runForeverThrottled logfun name burst rate a = mask $ \umask -> do
     void go `finally` logfun Info (name <> " stopped")
 
 -- -------------------------------------------------------------------------- --
--- Count leading zeros of a bytestring
-
--- | Count leading zeros of a bytestring
---
-leadingZeros :: B.ByteString -> Natural
-leadingZeros b = int (B.length x) * 8 + case B.uncons y of
-    Just (h, _) -> int $ countLeadingZeros h
-    Nothing -> 0
-  where
-    (x, y) = B.span (== 0x00) b
+-- Count leading zero bits of a bytestring
+
+-- | Count leading zero bits of a bytestring
+--
+leadingZeros :: Integral int => B.ByteString -> int
```

Comment:
I should probably move this hunk into its own PR -- I noticed it while I was reading this. Will delete